src/components/Locationitem/LocationItem.test.jsx

```diff
@@ -0,0 +1,21 @@
+import { describe, it, expect, vi } from "vitest";
+import { render, screen, fireEvent } from "@testing-library/react";
+import LocationItem from "./LocationItem";
+
+describe("<LocationItem />", () => {
+  it("renders the location name", () => {
+    render(<LocationItem text="New York" />);
+    expect(screen.getByText("New York")).toBeInTheDocument();
+  });
+
+  it("calls handleLocation when clicked", () => {
+    //creates a mock function
+    const handleLocation = vi.fn();
+    render(<LocationItem text="New York" handleLocation={handleLocation} />);
+
+    //simulates a click on the item
+    fireEvent.click(screen.getByText("New York"));
+
+    expect(handleLocation).toHaveBeenCalledTimes(1);
```

Comment:
When it's only one time, you can use `toHaveBeenCalledOnce` and it does the same:

```suggestion
    expect(handleLocation).toHaveBeenCalledOnce();
```